tests/test_awsdatabricks.py

```diff
@@ -0,0 +1,625 @@
+                    "EnaSupport": True,
+                    "Hypervisor": "xen",
+                    "InstanceLifecycle": "spot",
+                    "NetworkInterfaces": [],
+                    "RootDeviceName": "/dev/sda1",
+                    "RootDeviceType": "ebs",
+                    "SecurityGroups": [],
+                    "SpotInstanceRequestId": "sir-mzqyjsdg",
+                    "StateReason": {
+                        "Code": "Client.UserInitiatedShutdown",
+                        "Message": "Client.UserInitiatedShutdown: User initiated shutdown",
+                    },
+                    "Tags": [
+                        {"Key": "RunName", "Value": "my_test_job"},
+                        {"Key": "Vendor", "Value": "Databricks"},
+                        {"Key": "management_service", "Value": "instance_manager_service"},
+                        {"Key": "ClusterId", "Value": "0101-214342-tpi6qdp2"},
+                        {"Key": "Creator", "Value": "user@domain.com"},
+                        {"Key": "JobId", "Value": "943021334099449"},
+                        {
+                            "Key": "Name",
+                            "Value": "workerenv-1187965937856149-f0bf0016-45bf-496b-9f05-16cf93ffb24d-worker",
+                        },
+                        {
+                            "Key": "ClusterName",
+                            "Value": "job-12345678910-run-75778-my_job_run",
+                        },
+                    ],
+                    "VirtualizationType": "hvm",
+                    "CpuOptions": {"CoreCount": 8, "ThreadsPerCore": 8},
+                    "CapacityReservationSpecification": {"CapacityReservationPreference": "open"},
+                    "HibernationOptions": {"Configured": False},
+                    "MetadataOptions": {
+                        "State": "pending",
+                        "HttpTokens": "required",
+                        "HttpPutResponseHopLimit": 2,
+                        "HttpEndpoint": "enabled",
+                        "HttpProtocolIpv6": "disabled",
+                        "InstanceMetadataTags": "disabled",
+                    },
+                    "EnclaveOptions": {"Enabled": False},
+                    "PlatformDetails": "Linux/UNIX",
+                    "UsageOperation": "RunInstances",
+                    "UsageOperationUpdateTime": datetime.fromisoformat("2023-04-04T22:51:24+00:00"),
+                    "MaintenanceOptions": {"AutoRecovery": "default"},
+                }
+            ]
+        },
+    ],
+}
+
+MOCK_DBX_CONF = DatabricksConf(
+    host="https://dbc-123.cloud.databricks.com",
+    token="my_secret_token",
+    aws_region_name="us-east-1",
+)
+
+
+def test_create_prediction_for_run_failed_run(respx_mock):
+    failure_response = {"error_code": "FAILED", "message": "This run failed"}
+
+    respx_mock.get("https://*.cloud.databricks.com/api/2.1/jobs/runs/get?run_id=75778").mock(
```

Comment:
@rmoneys updated all these tests to use `respx` to mock out our HTTP calls instead :+1:

Reply:
Love it!